core/src/main/kotlin/io/specmatic/core/pattern/AnyOfPattern.kt

```diff
@@ -0,0 +1,427 @@
+    }
+
+    override fun fillInTheBlanks(value: Value, resolver: Resolver, removeExtraKeys: Boolean): ReturnValue<Value> {
+        val patternToConsider = when (val resolvedPattern = resolveToPattern(value, resolver, this)) {
+            is ReturnFailure -> return resolvedPattern.cast()
+            else -> resolvedPattern.value
+        }
+        if (isPatternToken(value) && patternToConsider == this) return HasValue(resolver.generate(this))
+
+        val newPatterns = pattern.filter { it.typeAlias != null }.associateBy { it.typeAlias.orEmpty() }
+        val updatedResolver = resolver.copy(newPatterns = resolver.newPatterns.plus(newPatterns) ).updateLookupPath(this.typeAlias)
+
+        val results = pattern.asSequence().map { it.fillInTheBlanks(value, updatedResolver, removeExtraKeys) }
+        val successfulGeneration = results.firstOrNull { it is HasValue }
+        if(successfulGeneration != null) return successfulGeneration
+
+        val resultList = results.toList()
+        val failures = resultList.filterIsInstance<ReturnFailure>().map { it.toFailure() }
+        return HasFailure(Failure.fromFailures(failures))
+    }
+
+    override fun resolveSubstitutions(
+        substitution: Substitution,
+        value: Value,
+        resolver: Resolver,
+        key: String?
+    ): ReturnValue<Value> {
+        val options = pattern.map {
+            try {
+                it.resolveSubstitutions(substitution, value, resolver, key)
+            } catch(e: Throwable) {
+                HasException(e)
+            }
+        }
+
+        val hasValue = options.find { it is HasValue }
+
+        if(hasValue != null)
+            return hasValue
+
+        val failures = options.map {
+            it.realise(
+                hasValue = { _, _ ->
+                    throw NotImplementedError()
+                },
+                orFailure = { failure -> failure.failure },
+                orException = { exception -> exception.toHasFailure().failure }
+            )
+        }
+
+        return HasFailure(Failure.fromFailures(failures))
+    }
+
+    override fun getTemplateTypes(key: String, value: Value, resolver: Resolver): ReturnValue<Map<String, Pattern>> {
+        val initialValue: ReturnValue<Map<String, Pattern>> = HasValue(emptyMap())
+
+        return pattern.fold(initialValue) { acc, pattern ->
+            val templateTypes = pattern.getTemplateTypes("", value, resolver)
+            acc.assimilate(templateTypes) { data, additional -> data + additional }
+        }
+    }
+
+    override fun matches(sampleData: Value?, resolver: Resolver): Result {
+        val matchResults: List<AnyOfPatternMatch> =
+            pattern.map {
+                AnyOfPatternMatch(it, resolver.matchesPattern(key, it, sampleData ?: EmptyString))
+            }
+
+        val matchResult = matchResults.find { it.result is Result.Success }
+
+        if(matchResult != null)
+            return matchResult.result
+
```

Comment:
If success match is not found, just return Result.fromFailures(matchResults). No further processing.